### ISSUE TITLE
Add a TextResponse?

### ISSUE THREAD

**Issue by mnapoli** [zendframework/zend-diactoros#76]
There is currently the base `Response`, but also `HtmlResponse`, `JsonResponse`, etc.

That's practical when we want to return HTML or JSON, but there is no simple `TextResponse` (or `PlainTextResponse`. That's not a huge deal, but writing a simple *hello world* is just plain confusing.

Example:

```php
// Super verbose
$response = new Response();
$res = $res->withHeader('content-type', 'text/plain');
$response->getBody()->write('Hello world!');
return $response;

// Wrong content type, but at least it's simpler and shorter
return new HtmlResponse('Hello world');
```

Do you think it would make sense to add a new `TextResponse` class?

Or even simpler if we could use the base Response class to return plain text content, but the $body parameter in the constructor seems to only accept streams. Maybe it could accept a plain string? (but then how to differentiate from a stream…)

**Comment by weierophinney** [comment]
@mnapoli The base `Response` class is supposed to be general-purpose, and work for both client and server-side; as such, the constructor targets a stream. (As you note, this also prevents the need for trying to differentiate from a stream; it's either a valid stream resource DSN or it isn't.)

The various response types under the `Zend\Diactoros\Response` hierarchy are the more specific implementations, and I'd be quite happy to entertain a `TextResponse` or `PlainTextResponse` implementation there.

**Comment by mnapoli** [comment]
:+1: makes sense about the generic response class, will work on a TextResponse tomorrow!

**Comment by weierophinney** [comment]
Closed with #77.
